airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/mongodb/MongoDbCdcTargetPosition.java

```diff
@@ -91,14 +93,29 @@ public BsonTimestamp extractPositionFromHeartbeatOffset(final Map<String, ?> sou
 
   @Override
   public boolean isEventAheadOffset(final Map<String, String> offset, final ChangeEventWithMetadata event) {
+    if (offset.size() != 1) {
+      return false;
+    }
+
     return MongoDbResumeTokenHelper.extractTimestampFromEvent(event.eventValueAsJson()).getValue() >= ResumeTokens
-        .getTimestamp(ResumeTokens.fromData(offset.get(MongoDbDebeziumConstants.OffsetState.VALUE_RESUME_TOKEN))).getValue();
+        .getTimestamp(ResumeTokens.fromData(getResumeToken(offset))).getValue();
   }
 
   @Override
   public boolean isSameOffset(final Map<String, String> offsetA, final Map<String, String> offsetB) {
-    return offsetA.get(MongoDbDebeziumConstants.OffsetState.VALUE_RESUME_TOKEN)
-        .equals(offsetB.get(MongoDbDebeziumConstants.OffsetState.VALUE_RESUME_TOKEN));
+    if (offsetA == null || offsetA.size() != 1) {
+      return false;
+    }
+    if (offsetB == null || offsetB.size() != 1) {
+      return false;
+    }
+
+    return getResumeToken(offsetA).equals(getResumeToken(offsetB));
+  }
+
```

Comment:
@jpefaur I don't think so.  I think there is some confusion here because of the name of the input parameters.  In the case of the `isEventAheadOfOffset` method, that method receives the map stored in the offset file, which is different from the `sourceOffset` in this method, which is a map of the value in the input to the `isEventAheadOfOffset`.